type_def.h

```diff
@@ -0,0 +1,8 @@
+enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
```

Comment:
The definition of colors is here, but their names are elsewhere. Any way to make it intuitive to update them together when a color gets added?